build-scripts/build_tests.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import pathlib
+from typing import TypeVar, List, Iterable, Set, Dict
+import multiprocessing
+
+import ssg.environment
+import ssg.jinja
+import ssg.utils
+import ssg.yaml
+import ssg.templates
+import tests.ssg_test_suite.common
+import tests.ssg_test_suite.rule
+
+SSG_ROOT = str(pathlib.Path(__file__).resolve().parent.parent.absolute())
+JOB_COUNT = multiprocessing.cpu_count()
+T = TypeVar('T')
+
+
+def _create_arg_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description="Converts built content tests to be rendered."
+    )
+    parser.add_argument(
+        "--build-config-yaml", required=True,
+        help="YAML file with information about the build configuration. "
+             "e.g.: ~/scap-security-guide/build/build_config.yml"
+    )
+    parser.add_argument(
+        "--product-yaml", required=True,
+        help="YAML file with information about the product we are building. "
+             "e.g.: ~/scap-security-guide/rhel10/product.yml"
+    )
+    parser.add_argument(
+        "--output", required=True,
+        help="Output path"
+             "e.g.:  ~/scap-security-guide/build/rhel10/tests"
+    )
+    parser.add_argument(
+        "--resolved-rules-dir", required=True,
+        help="Directory with <rule-id>.yml resolved rule YAMLs"
+             "e.g.: ~/scap-security-guide/build/rhel10/rules"
+    )
+    parser.add_argument("--log-level", action="store", type=str, default="ERROR",
+                        choices=["ERROR", "WARNING", "INFO", "DEBUG", "TRACE"],
+                        help="What level to log at. Defaults to ERROR.")
+    parser.add_argument("--root", default=SSG_ROOT,
+                        help=f"Path to the project. Defaults to {SSG_ROOT}")
+    parser.add_argument("--jobs", "-j", type=int, default=JOB_COUNT,
+                        help=f"Number of cores to use. Defaults to {JOB_COUNT} on this system.")
+    return parser
+
+
+def _is_test_file(filename: str) -> bool:
+    return filename.endswith(('.pass.sh', '.fail.sh', '.notapplicable.sh'))
+
+
+def _get_deny_templated_scenarios(test_config_path):
+    deny_templated_scenarios = list()
+    if test_config_path.exists():
+        test_config = ssg.yaml.open_raw(str(test_config_path.absolute()))
+        if 'deny_templated_scenarios' in test_config:
+            deny_templated_scenarios = test_config['deny_templated_scenarios']
+    return deny_templated_scenarios
+
+
+def _process_shared_file(env_yaml, file, shared_output_path):
+    file_contents = ssg.jinja.process_file_with_macros(str(file.absolute()),
+                                                       env_yaml)
+    shared_script_path = shared_output_path / file.name
+    with open(shared_script_path, 'w') as file:
+        file.write(file_contents)
+        file.write('\n')
```

Comment:
```suggestion
    _write_path(file_contents, shared_script_path)
```
?

(And in other similar cases.)